### PR TITLE
Prioritise filter sequence and visibility #961

### PR DESCRIPTION
**Pull request by verheyenkoen**
This PR adds UX features to facet filters for curators in the publications view as described in #961.

### CODE REVIEW

File: models/search_args.go [hunk 1]
```diff
@@ -1,12 +1,21 @@
...
-	Sort     []string            `query:"sort,omitempty"`
-	PageSize int                 `query:"page-size"`
-	Facets   []string            `query:"-"`
+	Query      string              `query:"q,omitempty"`
+	Filters    map[string][]string `query:"f,omitempty"`
+	Page       int                 `query:"page"`
+	Sort       []string            `query:"sort,omitempty"`
+	PageSize   int                 `query:"page-size"`
+	Facets     []string            `query:"-"`
+	FacetLines [][]string          `query:"-"`
```
Comment (verheyenkoen) [inline models/search_args.go:18]: Not sure if this `query:"-"` annotation at the end is right here (or what it does).

File: views/search/_facets.gohtml [hunk 2]
```diff
@@ -1,27 +1,64 @@
...
-                        <i class="if if-close"></i>
-                        <span class="btn-text">Reset filters</span>
-                    </button>
-                </form>
-            </div>
+  </div>
+  <div class="bc-toolbar-right">
+    <div class="bc-toolbar-item">
+      <div class="c-button-toolbar">
+        {{if gt (len .SearchArgs.FacetLines) 2}}
```
Comment (verheyenkoen) [inline views/search/_facets.gohtml:20]: I created a constant const for this number of lines (2) [here](https://github.com/ugent-library/biblio-backoffice/blob/7f30bcb376ef31bb76d37533af9f866a06c3c422/models/search_args.go#L9) but not sure how to use that in gohtml. Probably would need to pass that on with the context?

**Comment by verheyenkoen** [comment]
@mietcls Got a working version that you can maybe already have a look at and provide feedback for. For now I'll leave this draft but it is testable if you want.

**Review by nics — APPROVED** [review]

File: views/search/_facets.gohtml (hunk 2 above)
Comment (nics) [reply views/search/_facets.gohtml:20]: we clean that up after the templ rewrite

**Pull request merged**